Mapping/Loader/YamlLoader.php

```diff
@@ -0,0 +1,167 @@
+
+                if (!isset($propertyConfig['type']) && !is_scalar($propertyConfig['type'])) {
+                    throw new \RuntimeException('The "type" attribute must be set for each property.');
+                }
+
+                if ((string) $propertyConfig['type'] === 'attribute') {
+                    $field = new Property();
+                    $field->setName($property->getName());
+
+                    if (isset($propertyConfig['algoliaName'])) {
+                        $field->setAlgoliaName($propertyConfig['algoliaName']);
+                    } else {
+                        $field->setAlgoliaName($field->getName());
+                    }
+
+                    $description->addProperty($field);
+                } elseif ((string) $propertyConfig['type'] === 'id') {
+                    $description->addIdentifierAttributeName($propertyName);
+                } else {
+                    throw new \RuntimeException('The "type" attribute is wrongly configured.');
+                }
+            }
+        }
+    }
+
+    /**
+     * @param \ReflectionClass $class
+     * @param $className
+     * @param $config
+     * @param $description
+     */
+    protected function extractMethodMetadata(\ReflectionClass $class, $className, $config, Description $description)
+    {
+        foreach ($class->getMethods() as $method) {
+            if ($className !== $method->class) {
+                continue;
+            }
+
+            $methodName = $method->getName();
+
+            if (isset($config['methods'][$methodName])) {
+                $methodConfig = $config['methods'][$methodName];
+
+                if (!isset($methodConfig['type']) && !is_scalar($methodConfig['type'])) {
+                    throw new \RuntimeException('The "type" attribute must be set for each methods.');
+                }
+
+                if ((string) $methodConfig['type'] === 'attribute') {
+                    $field = new Method();
+                    $field->setName($methodName);
+
+                    if (isset($methodConfig['algoliaName'])) {
+                        $field->setAlgoliaName($methodConfig['algoliaName']);
+                    } else {
+                        $field->setAlgoliaName(lcfirst(preg_replace('/^get([A-Z])/', '$1', $field->getName())));
+                    }
+
+                    $description->addMethod($field);
+                } elseif ((string) $methodConfig['type'] === 'indexIf') {
+                    $indexIf = new IndexIf();
+                    $indexIf->setName($methodName);
+
+                    $description->addIndexIf($indexIf);
+                } else {
+                    throw new \RuntimeException('The "type" attribute is wrongly configured on.');
+                }
+            }
+        }
+    }
+
+    private function removeProxy($class)
+    {
+        /* Avoid proxy class form symfony */
+        return str_replace('Proxies\\__CG__\\', '', $class);
```

Comment:
This is wrong in a few other places as well, but you should always use `Doctrine\Common\Util\ClassUtils::getRealClass` to get the real class name.